src/NUnit.Portable.Agent/ResultSummary.cs

```diff
@@ -102,8 +102,14 @@ public XDocument GetTestResults()
             test.Add(new XAttribute("duration", Duration.ToString("0.000000", NumberFormatInfo.InvariantInfo)));
 
             foreach (var result in _results)
-                test.Add(result);
-            
+            {
+                XElement firstSuite = result.Name.LocalName == "test-run"
+                    ? result.Element(XName.Get("test-suite"))
+                    : result;
+
```

Comment:
I think there could be a need for a more fundamental change here. Why, at this stage of the execution pipeline, is there a test-run element at all.

The test-run element is only ever created in the engine at the very top level. It's like decorating a cake - you don't do it till the layers are assembled.